ReSharper.FSharp/src/FSharp/FSharp.Psi.Daemon/src/Highlightings/TypeHintHighlighting.fs

```diff
@@ -24,11 +25,32 @@ open JetBrains.UI.RichText
      AttributeId = AnalysisHighlightingAttributeIds.PARAMETER_NAME_HINT,
      OverlapResolve = OverlapResolveKind.NONE,
      ShowToolTipInStatusBar = false)>]
-type TypeHintHighlighting(typeNameString: string, range: DocumentRange, pushToHintMode: PushToHintMode, suffix,
-                          bulbActionsProvider: IInlayHintBulbActionsProvider, owner: ITreeNode) =
-    let text = RichText(": " + typeNameString + suffix)
+type TypeHintHighlighting private (typeString: string, owner: ITreeNode, range: DocumentRange,
+                                   pushToHintMode: PushToHintMode, bulbActionsProvider: IInlayHintBulbActionsProvider,
+                                   suffix) =
+    let text = RichText(": " + typeString + suffix)
+
     new (typeNameString: string, range: DocumentRange) =
-        TypeHintHighlighting(typeNameString, range, PushToHintMode.Default, "", null, null)
+        TypeHintHighlighting(typeNameString, null, range, PushToHintMode.Default, null, "")
+
+    new (typeNameString: string, binding: IBinding, pushToHintMode: PushToHintMode, bulbActionsProvider: IInlayHintBulbActionsProvider) =
+        let range =
+            if binding.HasParameters then binding.EqualsToken.GetDocumentRange().StartOffsetRange()
```

Comment:
Do we need check `EqualsToken` for `null` somewhere? Could you add some tests with unfinished bindings, if there aren't any yet, please?

Reply:
Cool, thank you!